examples/local_sgd/resnet/train_with_fleet.py

```diff
@@ -434,115 +484,157 @@ def if_exist(var):
+                all_train_time.append(train_end - train_begin)
+                train_speed = (batch_id * train_batch_size) / (train_end - train_begin)
+                print("current local_sgd_steps {}".format(local_sgd_steps))
+                print("End pass {0}, train_loss {1}, train_acc1 {2}, train_acc5 {3}, "
+                      "speed {4}".format(\
+                      pass_id, "%.5f"%train_loss, "%.5f"%train_acc1, "%.5f"%train_acc5, "%.2f" % train_speed))
+                sys.stdout.flush()
+                #init 
+                batch_id = 0
+                train_info = [[], [], []]
+                train_begin=time.time()
+                batch_id = 0
+                time_record=[]
+
+                if args.isTest:#test
+                    test_info = [[], [], []]
+                    test_py_reader.start()
+                    test_batch_id = 0
+                    try:
+                        while True:
+                            t1 = time.time()
+                            loss, acc1, acc5 = exe.run(program=test_prog,\
+                                                       fetch_list=test_fetch_list)
+                            t2 = time.time()
+                            period = t2 - t1
+                            loss = np.mean(loss)
+                            acc1 = np.mean(acc1)
+                            acc5 = np.mean(acc5)
+                            test_info[0].append(loss)
+                            test_info[1].append(acc1)
+                            test_info[2].append(acc5)
+
+                            if test_batch_id % 200 == 0:
+                                test_speed = test_batch_size * 1.0 / period
+                                print("Pass {0},testbatch {1},loss {2}, acc1 {3},acc5 {4},time {5},speed {6}"\
+                                     .format(pass_id, test_batch_id, "%.5f"%loss,"%.5f"%acc1, "%.5f"%acc5,\
+                                     "%2.2f sec" % period, "%.2f" % test_speed))
+                                sys.stdout.flush()
+                            test_batch_id += 1
+                    except fluid.core.EOFException:
+                        test_py_reader.reset()
+
+                    test_loss = np.array(test_info[0]).mean()
+                    test_acc1 = np.array(test_info[1]).mean()
+                    test_acc5 = np.array(test_info[2]).mean()
+
+                    print("End pass {0}, test_loss {1}, test_acc1 {2}, test_acc5 {3}".format(pass_id,"%.5f"%test_loss,\
+                          "%.5f"%test_acc1, "%.5f"%test_acc5))
+                    sys.stdout.flush()
+    except fluid.core.EOFException:
+        train_py_reader.reset()
+
+    #start test
+    test_py_reader.start()
+    test_batch_id = 0
+    test_info = [[], [], []]
+    try:
+        while True:
+            t1 = time.time()
+            loss, acc1, acc5 = exe.run(program=test_prog,\
+                                       fetch_list=test_fetch_list)
```

Comment:
u should better help creating a distributed prediction algorithm in the furture.